cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -106,6 +107,17 @@ func clientForUserAgentOrDie(config client.Config, userAgent string) *client.Cli
 	return kubeClient
 }
 
+// clientsetForUserAgentOrDie is going to replace clientForUserAgentOrDie
+func clientsetForUserAgentOrDie(config client.Config, userAgent string) release_1_1.Interface {
```

Comment:
could this be factored as a function that copies the config and sets the user agent, and then calling a standard clientsetFromConfig function somewhere?